packages/host/app/components/definition-container.gts

```diff
@@ -0,0 +1,204 @@
+import Label from '@cardstack/boxel-ui/components/label';
+import Component from '@glimmer/component';
+import { on } from '@ember/modifier';
+import { action } from '@ember/object';
+import { svgJar } from '@cardstack/boxel-ui/helpers/svg-jar';
+import { Button } from '@cardstack/boxel-ui';
+import { eq, and } from '@cardstack/boxel-ui/helpers/truth-helpers';
+// boxel- specific
+import { type RealmInfo } from '@cardstack/runtime-common';
+
+export type DefinitionVariant = 'instance' | 'module';
+
+interface Signature {
+  Element: HTMLElement;
+  Args: {
+    name: string | undefined;
+    fileExtension: string;
+    realmInfo: RealmInfo | null;
+    infoText: string | undefined;
+    variant: DefinitionVariant;
+    isActive: boolean;
+    onCreate?: () => void;
+    onInherit?: () => void;
+    onDuplicate?: () => void;
+  };
+}
+
+export default class DefinitionContainer extends Component<Signature> {
+  @action
+  duplicateAction() {
+    if (this.args.onDuplicate) {
+      this.args.onDuplicate();
+    }
+  }
+
+  @action
+  createAction() {
+    if (this.args.onCreate) {
+      this.args.onCreate();
+    }
+  }
+
+  @action
+  inheritAction() {
+    if (this.args.onInherit) {
+      this.args.onInherit();
+    }
+  }
+
+  get title(): string {
+    switch (this.args.variant) {
+      case 'module':
+        return 'CARD DEFINITION';
+      case 'instance':
+        return 'CARD INSTANCE';
+      default:
+        return 'CARD DEFINITION';
```

Comment:
We could use `assertNever` approach like in this line https://github.com/cardstack/boxel/blob/39c0d13151e7c639c8c7cb1aa2cf24c8bcb2468a/packages/host/app/components/operator-mode/container.gts#L607.